ui/pages/confirmations/components/confirm/info/approve/hooks/use-is-nft.ts

```diff
@@ -0,0 +1,23 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { useEffect, useState } from 'react';
+import { getTokenStandardAndDetails } from '../../../../../../../store/actions';
+
+export const useIsNFT = (
+  transactionMeta: TransactionMeta,
+): { isNFT: boolean } => {
+  const [decimals, setDecimals] = useState('');
+  useEffect(() => {
+    const fetchTokenDetails = async () => {
+      const result = await getTokenStandardAndDetails(
+        transactionMeta?.txParams?.to as string,
+      );
+
+      setDecimals(result?.decimals as string);
+    };
+
+    fetchTokenDetails();
+  }, [transactionMeta]);
+  const isNFT = decimals === '0';
```

Comment:
Is this a safe assumption, could an ERC 20 choose to use zero decimals?

Should we instead check the standard returned in the response?